nodes/generators_extended/spiral_mk2.py

```diff
@@ -0,0 +1,655 @@
+
+    height: FloatProperty(
+        name="Height", description="Height of the spiral along z",
+        default=0.0, update=update_spiral)
+
+    phase: FloatProperty(
+        name="Phase", description="Phase amount around spiral center",
+        default=0.0, update=SvAngleHelper.update_angle)
+
+    exponent: FloatProperty(
+        name="Exponent", description="Exponent attenuator",
+        default=2.0, update=update_spiral)
+
+    resolution: IntProperty(
+        name="Turn Resolution", description="Number of vertices in one turn in the spiral",
+        default=100, min=3, update=update_spiral)
+
+    separate: BoolProperty(
+        name="Separate arms",
+        description="Separate the spiral arms",
+        default=False, update=update_spiral)
+
+    updating: BoolProperty(default=False)  # used for disabling update callback
+
+    def migrate_from(self, old_node):
+        ''' Migration from old nodes '''
+        if old_node.bl_idname == "SvSpiralNode":
+            self.sType = old_node.stype
+            self.last_angle_units = AngleUnits.RADIANS
+            self.angle_units = AngleUnits.RADIANS
+
+    def sv_init(self, context):
+        self.width = 170
+        self.inputs.new('SvStringsSocket', "R").prop_name = 'eRadius'
+        self.inputs.new('SvStringsSocket', "r").prop_name = 'iRadius'
+        self.inputs.new('SvStringsSocket', "e").prop_name = 'exponent'
+        self.inputs.new('SvStringsSocket', "t").prop_name = 'turns'
+        self.inputs.new('SvStringsSocket', "n").prop_name = 'resolution'
+        self.inputs.new('SvStringsSocket', "s").prop_name = 'scale'
+        self.inputs.new('SvStringsSocket', "h").prop_name = 'height'
+        self.inputs.new('SvStringsSocket', "p").prop_name = 'phase'
+        self.inputs.new('SvStringsSocket', "a").prop_name = 'arms'
+
+        self.outputs.new('SvVerticesSocket', "Vertices")
+        self.outputs.new('SvStringsSocket', "Edges")
+
+        self.presets = "ARCHIMEDEAN"
+
+    def draw_buttons(self, context, layout):
+        layout.prop(self, 'presets')
+        layout.prop(self, 'sType', text="")
+        col = layout.column(align=True)
+        if self.sType in ("LOGARITHMIC", "ARCHIMEDEAN", "SPIRANGLE"):
+            row = col.row(align=True)
+            row.prop(self, 'normalize', expand=True)
+        row = col.row(align=True)
+        row.prop(self, 'flip', text="Flip", toggle=True)
+        row.prop(self, 'separate', text="Separate", toggle=True)
+
+    def draw_buttons_ext(self, context, layout):
+        self.draw_angle_units_buttons(context, layout)
+
+    def process(self):
+        outputs = self.outputs
+        # return if no outputs are connected
+        if not any(s.is_linked for s in outputs):
+            return
+
+        # input values lists (single or multi value)
+        inputs = self.inputs
+        input_R = inputs["R"].sv_get()[0]  # list of exterior radii
+        input_r = inputs["r"].sv_get()[0]  # list of interior radii
```

Comment:
Using capitalized letters in names of variables is against Python convention.

Reply:
you’re right. I did **most** of the conversion. I only left these R/r ones since I didn’t want to add longer names like input_major_radius & input_minor_radius. it’s not gonna be 100% perfect. Much of the SV code is not 100% compliant at this point. e.g. updateNode() will unlikely change since it’s so pervasive.